pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -40,62 +40,32 @@ type ExecutionStartInfo struct {
 	SamplingPriority sampler.SamplingPriority
 }
 
-type invocationPayload struct {
-	Headers map[string]string `json:"headers"`
-}
-
 // startExecutionSpan records information from the start of the invocation.
 // It should be called at the start of the invocation.
-func (lp *LifecycleProcessor) startExecutionSpan(rawPayload []byte, startDetails *InvocationStartDetails) {
-	payload := convertRawPayload(rawPayload)
-	inferredSpan := lp.GetInferredSpan()
+func (lp *LifecycleProcessor) startExecutionSpan(event interface{}, rawPayload []byte, startDetails *InvocationStartDetails) {
 	executionContext := lp.GetExecutionInfo()
 	executionContext.requestPayload = rawPayload
 	executionContext.startTime = startDetails.StartTime
 
-	if lp.InferredSpansEnabled && inferredSpan.Span.Start != 0 {
-		executionContext.TraceID = inferredSpan.Span.TraceID
-		executionContext.parentID = inferredSpan.Span.SpanID
+	traceContext, err := lp.Extractor.Extract(event, rawPayload)
+	if err != nil {
+		traceContext = lp.Extractor.ExtractFromLayer(startDetails.InvokeEventHeaders).TraceContext
 	}
 
-	if payload.Headers != nil {
-
-		traceID, err := strconv.ParseUint(payload.Headers[TraceIDHeader], 0, 64)
-		if err != nil {
-			log.Debug("Unable to parse traceID from payload headers")
-		} else {
-			executionContext.TraceID = traceID
-			if lp.InferredSpansEnabled {
-				inferredSpan.Span.TraceID = traceID
-			}
-		}
-
-		parentID, err := strconv.ParseUint(payload.Headers[ParentIDHeader], 0, 64)
-		if err != nil {
-			log.Debug("Unable to parse parentID from payload headers")
-		} else {
-			if lp.InferredSpansEnabled {
-				inferredSpan.Span.ParentID = parentID
-			} else {
-				executionContext.parentID = parentID
-			}
-		}
-	} else if startDetails.InvokeEventHeaders.TraceID != "" { // trace context from a direct invocation
-		traceID, err := strconv.ParseUint(startDetails.InvokeEventHeaders.TraceID, 0, 64)
-		if err != nil {
-			log.Debug("Unable to parse traceID from invokeEventHeaders")
-		} else {
-			executionContext.TraceID = traceID
-		}
-
-		parentID, err := strconv.ParseUint(startDetails.InvokeEventHeaders.ParentID, 0, 64)
-		if err != nil {
-			log.Debug("Unable to parse parentID from invokeEventHeaders")
-		} else {
-			executionContext.parentID = parentID
+	if traceContext != nil {
+		executionContext.TraceID = traceContext.TraceID
```

Comment:
I think we still need to check for the inferred span here. If inferred span is enabled. Then we need to set the 
```go
inferredSpan.Span.TraceID = traceContext.TraceID
inferredSpan.Span.ParentID = traceContext.ParentID
executionContext.TraceID = inferredSpan.Span.TraceID
executionContext.parentID = inferredSpan.Span.SpanID
```